src/main/java/org/reactome/addlinks/referencecreators/XenbaseReferenceCreator.java

```diff
@@ -0,0 +1,71 @@
+package org.reactome.addlinks.referencecreators;
+
+import org.gk.model.GKInstance;
+import org.gk.model.ReactomeJavaConstants;
+import org.gk.persistence.MySQLAdaptor;
+
+import java.util.List;
+import java.util.Map;
+
+public class XenbaseReferenceCreator extends SimpleReferenceCreator<List<String>>{
+    public XenbaseReferenceCreator(MySQLAdaptor adapter, String classToCreate, String classReferring, String referringAttribute, String sourceDB, String targetDB)
+    {
+        super(adapter, classToCreate, classReferring, referringAttribute, sourceDB, targetDB);
+    }
+
+    public XenbaseReferenceCreator(MySQLAdaptor adapter, String classToCreate, String classReferring, String referringAttribute, String sourceDB, String targetDB, String refCreatorName)
+    {
+        super(adapter, classToCreate, classReferring, referringAttribute, sourceDB, targetDB, refCreatorName);
+    }
+
+    @Override
+    public void createIdentifiers(long personID, Map<String, List<String>> mapping, List<GKInstance> sourceReferences) throws Exception
+    {
+        int sourceIdentifiersWithNoMapping = 0;
+        int sourceIdentifiersWithNewIdentifier = 0;
+        int sourceIdentifiersWithExistingIdentifier = 0;
+
+        for (GKInstance sourceInst : sourceReferences)
+        {
+            String sourceIdentifier = (String) sourceInst.getAttributeValue(ReactomeJavaConstants.identifier);
+            Long species = ((GKInstance) sourceInst.getAttributeValue(ReactomeJavaConstants.species)).getDBID();
+
+            if (mapping.containsKey(sourceIdentifier))
+            {
+                for (String xenbaseID : mapping.get(sourceIdentifier))
+                {
+                    if (!this.checkXRefExists(sourceInst, xenbaseID))
+                    {
+                        sourceIdentifiersWithNewIdentifier ++;
+                        if (!this.testMode)
+                        {
+                            this.refCreator.createIdentifier(xenbaseID, String.valueOf(sourceInst.getDBID()), targetRefDB, personID, this.getClass().getName(), species);
+                        }
+                    }
+                    else
+                    {
+                        sourceIdentifiersWithExistingIdentifier ++;
+                    }
+                }
+            }
+            else
+            {
+                sourceIdentifiersWithNoMapping ++;
+            }
+        }
+        // If some of these numbers don't seem to quite add up, remember that sourceIdentifiersWithNewIdentifier
+        // (and possibly sourceIdentifiersWithExistingIdentifier) will be incremented in a loop over HGNC IDs.
```

Comment:
"HGNC IDs"...? Copy and paste, eh? ;) Might want to check the other Reference Creators you've created. ;)

Reply:
Woopsies -- yep you caught me!